geoips/geoips_utils.py

```diff
@@ -75,40 +75,30 @@ def load_all_yaml_plugins():
     in ``.yaml``. Read each plugin file
     """
     # Load all entry points for plugin packages
-    plugin_packages = get_entry_point_group("geoips.plugin_packages")
 
-    # Loop over the plugin packages and load all of their yaml plugins
+    reg_plug_path = resources.files("geoips") / "registered_plugins.yaml"
+    if not os.path.exists(reg_plug_path):
+        from . import create_plugin_registry
+
+        create_plugin_registry.main()
+    plugin_packages = get_entry_point_group("geoips.plugin_packages")
     plugins = {}
+    yaml_interfaces = [
+        "feature_annotators",
+        "gridline_annotators",
+        "product_defaults",
+        "products",
+        "sectors",
+    ]
```

Comment:
@evrose54 @jsolbrig  Is there a way to programmatically determine this, without an explicit list of the yaml interfaces ?  I'm not sure there is easily.

Reply:
https://github.com/NRLMMD-GEOIPS/geoips/pull/240/files#r1373462636
^ I fixed it!